### PR TITLE
Where-Object: add parameter 'Not'

### LINKED ISSUE #3529: Add Where-Object -Not support
I often have the need to filter for objects that do not have a property specified or who have a falsey value for a property by doing something like the following:

```
$Computers = [PSCustomObject]@{
    ComputerName = "SPC-1234"
    IPAddress = "192.168.0.1"
},
[PSCustomObject]@{
    ComputerName = "BGP-5678"
    IPAddress = ""
},
[PSCustomObject]@{
    ComputerName = "MGC-9101"
}

$Computers | 
Where {-not $_.IPAddress }
```
```
ComputerName IPAddress
------------ ---------
BGP-5678              
MGC-9101 
```

If I am doing the opposite test I can avoid using a script block with Where-Object:

```
$Computers |
Where IPAddress
```

```
ComputerName IPAddress  
------------ ---------  
SPC-1234     192.168.0.1
```

I would like to be able to also avoid using a script block when using -Not which would enable the following:
```
$Computers | 
where -not IPAddress
```

### PR DESCRIPTION
**Pull request by SimonWahlin**
## PR Summary

Add parameter Not to Where-Object

Allows following example:
`Get-Service | Where-Object -Not DependentServices`
Returns all services without any dependent services.

Fix #3529 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed - Issue link: PowerShell/PowerShell-Docs#2199
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/InternalCommands.cs [hunk 1]
```diff
@@ -1199,6 +1200,16 @@ public SwitchParameter IsNot
             get { return _binaryOperator == TokenKind.IsNot; }
         }
 
+        /// <summary>
+        /// Binary operator -Not
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/InternalCommands.cs]: Please add final dot in the public comment.

File: test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+Describe "Where-Object" -Tags "CI" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1]: Please insert blank line after the copyright header.

File: test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                ComputerName = "MGC-9101"
+                NumberOfCores = 3
+                Drives = 'C'
+            }
+        )
+    }
+
+    It "Where-Object -Not Prop" {
+        $Result = $Computers | Where-Object -Not 'IPAddress'
+        $Result.Count | Should -BeExactly 2
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1]: We use `-BeExactly` only for strings. Please use `-Be`. Below too.

File: test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        $Result = $Computers | Where-Object -FilterScript {$_.NumberOfCores -lt 2}
+        $Result.Count | Should -BeExactly 1
+    }
+
+    It 'Where-Object Prop -lt 2' {
+        $Result = $Computers | Where-Object NumberOfCores -lt 2
+        $Result.Count | Should -BeExactly 1
+    }
+
+    # Like Match
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1]: Please remove the comment.

File: test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        $Result = $Computers | Where-Object -FilterScript {$_.ComputerName -match '^MGC.+'}
+        $Result.Count | Should -BeExactly 1
+    }
+
+    It 'Where-Object Prop -like Value' {
+        $Result = $Computers | Where-Object ComputerName -match '^MGC.+'
+        $Result.Count | Should -BeExactly 1
+    }
+}
+
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1]: Please remove extra line.

**Review by lzybkr — APPROVED** [review]

**Review by iSazonov — APPROVED** [review]
@SimonWahlin Thanks for your contribution!

**Review by daxian-dbw — APPROVED** [review]

**Comment by iSazonov** [comment]
@SimonWahlin I hope you continue push great contributions!

**Comment by chuanjiao10** [comment]
what to 'not' ???  It is  too ambiguous,
Not conducive to beginners, it is recommended to rename .

**Comment by iSazonov** [comment]
@chuanjiao10 Please open new Issue with your motivations and suggestion.

**Pull request merged**